### PR TITLE
Increase scope of docformatter and black [unitaryhack]

### LINKED ISSUE #66: Change scope of `docformatter` and `black` 👁️
_This issue has been tagged for a bounty during [unitaryHACK](https://unitaryfund.github.io/unitaryhack/participating-projects.html)_.

## Tasks
1. Include all Python files in this repository to be checked by `docformatter`
2. Include the `doc` directory in the scope of the `black` formatter.

## Explanation
Currently, we only check some `.py` files, namely `flamingpy/*/*.py`, with `docformatter` whereas it makes sense to check all Python files, including test files and files like `setup.py`. This will improve the quality of the docstrings and the code in general, making it easier to use and develop.

For part 1 of the task, you effectively have to change `python -m docformatter --in-place  flamingpy/*/*.py` to `python -m docformatter --in-place  **.py` in the CI checks. For part 2, make sure this does not break `doc` workflows!

## Submission
To work on this task, simply fork the repository. Please also comment in this thread to communicate that you are working on this issue. Once ready, you can open a PR that we will review and hopefully merge making you an official FlamingPy contributor 🔥 🥧 

Good luck 🍀 

\- Joost

### PR DESCRIPTION
**Pull request by WingCode**
## Context for changes

* Increase the scope of `docformatter` to all `.py` files in the repository. The behavior of `docformatter` has changed from checking and reporting incorrectly formatted docstring files to correct docstring in in-place.
* The scope of `black` has been extended to include `doc/` as well.

## Example usage and tests

- [x] Not Applicable

## Performance results justifying changes

- [x] Not Applicable

## Workflow actions and tests

- [x] Not Applicable

## Expected benefits and drawbacks

**Expected benefits:**
- Improved documentation quality since now it will be auto-formatted to the right format. 

**Possible drawbacks:**
- There can arise unexpected edge cases from auto-formatting the docs in-place. Thus, PR review needs to be more careful thereby might take more time.

## Related Github issues

- closes #66

## Checklist and integration statements

- [x] My Python and C++ codes follow this project's coding and commenting styles as indicated by existing files. Precisely, the changes conform to given `black`, `docformatter` and `pylint` configurations. 
- [x] I have performed a self-review of these changes, checked my code (including for [codefactor](https://www.codefactor.io/repository/github/xanaduai/flamingpy/branches) compliance), and corrected misspellings to the best of my capacity. I have synced this branch with others as required.

- [x] I have added context for corresponding changes in documentation and [`README.md`](README.md) as needed.
- [x] I have added new workflow CI tests for corresponding changes, ensuring codecoverage is 95% or better, and these pass locally for me.
- [x] I have updated [`CHANGELOG.md`](.github/CHANGELOG.md) following the template. I recognize that the developers may revisit [`CHANGELOG.md`](.github/CHANGELOG.md) and the versioning, and create a Special Release including my changes.

### CODE REVIEW

**Comment by soosub** [comment]
Don't forget part 2 of the task @WingCode 👀

**Comment by WingCode** [comment]
@soosub Regarding the part 2 of the task,
I checked in my local machine whether the change from `python -m docformatter --check flamingpy/*/*.py` to `python -m docformatter --in-place **.py` was breaking any documentation. It wasn't breaking any .py files and tests were successful.  

The command `python -m docformatter --in-place **.py`  was fixing [line 1](https://github.com/XanaduAI/flamingpy/blob/1516ff026a908bd636881e2c708f5293774596f8/setup.py#L14) and [line 2](https://github.com/XanaduAI/flamingpy/blob/1516ff026a908bd636881e2c708f5293774596f8/setup.py#L39) in place. Would like me to commit the fixed `setup.py` after `python -m docformatter --in-place **.py` as part of this PR?

**Comment by soosub** [comment]
> @soosub Regarding the part 2 of the task, I checked in my local machine whether the change from `python -m docformatter --check flamingpy/*/*.py` to `python -m docformatter --in-place **.py` was breaking any documentation. It wasn't breaking any .py files and tests were successful.
> 
> The command `python -m docformatter --in-place **.py` was fixing [line 1](https://github.com/XanaduAI/flamingpy/blob/1516ff026a908bd636881e2c708f5293774596f8/setup.py#L14) and [line 2](https://github.com/XanaduAI/flamingpy/blob/1516ff026a908bd636881e2c708f5293774596f8/setup.py#L39) in place. Would like me to commit the fixed `setup.py` after `python -m docformatter --in-place **.py` as part of this PR?

Ah, I think the second task might not have been clear. We would also like to increase the scope of the [`black`](https://github.com/psf/black) formatter in our repo (don't worry - it's not too much work). This is independent of the `docformatter`. I think it should be clear if you reread the corresponding issue #79. Hope this helps!

**Comment by soosub** [comment]
I'll convert the PR into a draft for now. Feel free to open it for review once it's ready!

**Comment by nariman87** [comment]
Just a general note that "**" is shell dependent and best is to use more general expression.

**Comment by WingCode** [comment]
@soosub My bad. Sorry missed that. I will make the change for `black` formatter
@nariman87 Good catch! How about we change to ` python -m docformatter -r --in-place .` since it by default looks out for .py files and thereby we don't have to define the globbing pattern.

File: .github/CHANGELOG.md
Comment (soosub) [inline .github/CHANGELOG.md:25]:
```suggestion
[Joost Bus](https://github.com/soosub), [Sebastián Duque Mesa](https://github.com/sduquemesa), [Luis Mantilla](https://github.com/BestQuark), [WingCode](https://github.com/WingCode)
```

**Comment by soosub** [comment]
> @soosub , @nariman87 Would like me to keep it as python -m docformatter -r --check . instead and could you let me know which files/folders which we would like to exclude from docformatter check/in-place?

A good rule of thumb is to always listen to @nariman87 😉 Indeed, let's go with `--check`. Regarding the scope, I think we can just include all `.py` files so what you did now is fine @WingCode. I'll open it for review 👍🏼

File: .github/workflows/format.yaml
Comment (soosub) [inline .github/workflows/format.yaml:34]:
```suggestion
        run: python -m docformatter --check **.py
```

**Comment by nariman87** [comment]
Yes, definitely only do --check. 


No particular file to exclude for now, we just need to review diffs carefully and if docformatter is applying an unintended change to some of the files, then exclude them.

**Review by nariman87 — APPROVED** [review]
Excellent work, WingCode, one approval from me.

Let's wait for @soosub's approval and possible further comments as well.

**Review by soosub — APPROVED** [review]
Looks good to me!

**Comment by ilan-tz** [comment]
Congratulations on resolving our first unitaryHACK issue, @WingCode ! I will fix some small CodeFactor issues from an earlier PR and then merge.

**Pull request merged**
